### PR TITLE
fix: don’t rebase if target branch changed

### LINKED ISSUE #4248: Bug: changing PR/MR target branch causes endless rebases
**What Renovate type are you using?**
Renovate Bot on GitLab

**Describe the bug**
Renovate Bot opened a merge request updating a dependency in my Node.js project to a new major version. Not wanting to land this on `master` yet, I retargeted the merge request to the `v6` branch where work on _my_ next major version is happening. Renovate Bot proceeded to get stuck in a loop committing the same package.json and package-lock.json changes over and over again until I closed the merge request.

**Did you see anything helpful in debug logs?**
No logs, but [here's the merge request in question](https://gitlab.com/dmfay/massive-js/merge_requests/710). The three commits landing at once before the spam starts seem of particular interest: I had been managing a few other Renovate merge requests within a few minutes of the incident, and one (which got merged into `master`) had updated the same dependency to a new patch version. It appears to have rolled this prior commit into the merge request, even though that commit should only appear on `master`.

**To Reproduce**
Uncertain, but here's what I suspect might have caused it:

1. Before anything happened, pg-promise was at 8.7.4 on both the `master` and `v6` branches.
2. I merged Renovate Bot's pull request updating pg-promise to 8.7.5 on `master`.
3. I retargeted Renovate Bot's pull request updating pg-promise to 9.0.0 on `v6`.

**Expected behavior**
Renovate Bot should have rebased its changes cleanly onto the new branch without including the new commits on `master` and without spamming the merge request.

### PR DESCRIPTION
**Pull request by rarkins**
Closes #4248

### CODE REVIEW

**Comment by rarkins** [comment]
@ViceIce do you know if Azure DevOps or either of the Bitbuckets let the user change the target branch similarly?

**Comment by ViceIce** [comment]
Changing target branch on pr is allowed on bitbucket server:
image

File: lib/platform/github/index.ts [hunk 1]
```diff
@@ -191,7 +191,11 @@ export async function initRepo({
   [config.repositoryOwner, config.repositoryName] = repository.split('/');
   config.gitPrivateKey = gitPrivateKey;
   // platformConfig is passed back to the app layer and contains info about the platform they require
-  const platformConfig: { privateRepo: boolean; isFork: boolean } = {} as any;
+  const platformConfig: {
```
Comment (ViceIce) [inline lib/platform/github/index.ts]: Should be `const platformConfig: PlatformConfig = {} as any`

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by ViceIce** [comment]
on azure too:
image

**Comment by ViceIce** [comment]
looks like we need to extend the `PlatformConfig` interface

image

**Comment by ViceIce** [comment]
for bitbucket server you can use `pr.toRef.displayId`

**Comment by ViceIce** [comment]
for azure `azurePr.targetRefName`

**Comment by ViceIce** [comment]
for bitubucket cloud `pr.destination.branch.name`

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**